server/sonar-server/src/test/java/org/sonar/server/usertoken/ws/GenerateActionTest.java

```diff
@@ -93,6 +97,15 @@ public void return_json_example() {
   }
 
   @Test
+  public void a_user_can_generate_token_for_himself() {
+    userSession.login(GRACE_HOPPER).setGlobalPermissions(GlobalPermissions.SCAN_EXECUTION);
+
+    GenerateWsResponse response = newRequest(null, TOKEN_NAME);
+
+    assertThat(response.getLogin()).isEqualTo(GRACE_HOPPER);
+  }
+
+  @Test
   public void fail_if_login_does_not_exist() {
     expectedException.expect(NotFoundException.class);
     expectedException.expectMessage("User with login 'unknown-login' not found");
```

Comment:
This is a vulnerability issue. If I'm not an administrator, I can know if an arbitrary login exists by calling /api/user_tokens/generate?name=test&login=foo. This request returns:
`{"errors":[{"msg":"User with login 'admin2' not found"}]}` whereas it should return `{"errors":[{"msg":"Insufficient privileges"}]}`